package/MDAnalysis/converters/OpenMMParser.py

```diff
@@ -103,41 +106,66 @@ def _mda_topology_from_omm_topology(self, omm_topology):
-        masses = [a.element.mass._value for a in omm_topology.atoms()]
         resnames = [r.name for r in omm_topology.residues()]
         resids = [r.index + 1 for r in omm_topology.residues()]
         resnums = resids.copy()
         segids = [str(c.index) for c in omm_topology.chains()]
         bonds = [(b.atom1.index, b.atom2.index) for b in omm_topology.bonds()]
         bond_orders = [b.order for b in omm_topology.bonds()]
         bond_types = [b.type for b in omm_topology.bonds()]
-
-        n_atoms = len(atomids)
-        n_residues = len(resids)
-        n_segments = len(segids)
+        elements = [None if a.element is None else a.element.symbol
+                    for a in omm_topology.atoms()]
 
         attrs = [
             Atomids(np.array(atomids, dtype=np.int32)),
             Atomnames(np.array(atomnames, dtype=object)),
-            Atomtypes(np.array(atomtypes, dtype=object)),
             Bonds(bonds, types=bond_types, order=bond_orders, guessed=False),
             ChainIDs(np.array(chainids, dtype=object)),
-            Elements(np.array(elements, dtype=object)),
-            Masses(np.array(masses, dtype=np.float32)),
             Resids(resids),
             Resnums(resnums),
             Resnames(resnames),
             Segids(segids),
         ]
 
+        if not any(elements):
+            atomtypes = guess_types(atomnames)
+            attrs.append(Atomtypes(np.array(atomtypes, dtype=object)))
+            masses = guess_masses(atomtypes)
+            attrs.append(Masses(np.array(masses)))
+            warnings.warn("Element information is missing, elements attribute "
+                          "will not be populated. If needed these can be "
+                          "guessed using MDAnalysis.topology.guessers.")
+        else:
+            validated_elements = []
+            masses = []
+            for a in omm_topology.atoms():
+                elem = a.element
+                if elem is not None and elem.symbol.capitalize() in SYMB2Z:
+                    validated_elements.append(a.element.symbol.capitalize())
+                    masses.append(a.element.mass._value)
+                else:
+                    wmsg = (f"Unknown element {elem} found for some atoms. "
+                            f"These have been given an empty element record. "
+                            f"and their mass has been set to 0.0 "
+                            f"If needed they can be guessed using "
+                            f"MDAnalysis.topology.guessers.")
+                    warnings.warn(wmsg)
+                    validated_elements.append('')
+                    masses.append(0.0)
+
+            attrs.append(Elements(np.array(validated_elements, dtype=object)))
+            attrs.append(Masses(np.array(masses)))
+            atomtypes = [elem for elem in validated_elements]
+            attrs.append(Atomtypes(np.array(atomtypes, dtype=object)))
```

Comment:
Currently in discussions with @jbarnoud but maybe we shouldn't be setting `atomtype` records to empty strings here. Since atomtypes are kind of a require attribute (for now), it's might make more sense to replace the empty positions with something like atomname or just any selectable character?

Thoughts @MDAnalysis/coredevs ?

Reply:
I would rather use a common type for all the non-assigned atoms. This way a user can select all of them at once. I would happily suggest going non-ascii here with something like ❌ but that might be controversial :stuck_out_tongue:

Reply:
> I would rather use a common type for all the non-assigned atoms. This way a user can select all of them at once. I would happily suggest going non-ascii here with something like x but that might be controversial stuck_out_tongue

I'd like the controversy :)  but settled for conventional `X` upon @IAlibay's request. Anyway thanks for the review, I've done the changes.

Reply:
> Quick question: afaik the PDB parser too uses empty string for absent elements. Should i make the changes there too?

The disprepency is annoying, indeed, but elements and types are different for various reasons. Changing how elements behave is likely to break downstream code so we will avoid doing it.

Reply:
I think there's a misunderstanding here - default elements should be '' and default atomtypes should be X.

The choice for '' in elements was the result of long discussions on the matter so I'd prefer not restart those at the moment.

Reply:
Apologies for the misunderstanding, I've made the corrections

> I think there's a misunderstanding here - default elements should be '' and default atomtypes should be X.
> 
> The choice for '' in elements was the result of long discussions on the matter so I'd prefer not restart those at the moment.

Apologies for the misunderstanding, I've made the corrections